app/src/main/java/com/example/pix/chat/adapters/MessageAdapter.java

```diff
@@ -82,17 +100,109 @@ public void bind(int position, List<Message> messages) {
             } catch (ParseException e) {
                 e.printStackTrace();
             }
+
+            // Only show text attachment if not a Snap content
             this.text.setVisibility(View.GONE);
-            if (message.getText() != null && !message.getText().equals("")) {
+            if (message.getText() != null && !message.getText().equals("") && !message.isSnap()) {
                 this.text.setVisibility(View.VISIBLE);
                 this.text.setText("" + message.getText());
             }
 
 
-            ParseFile pic = message.getPic();
-            if (pic != null) {
+            ParseFile content = message.getPic();
+            // Only show content attachment if not a Snap content
+            if (content != null && !message.isSnap()) {
                 this.contentPic.setVisibility(View.VISIBLE);
-                Glide.with(this.contentPics.getContext()).load(pic.getUrl()).into(this.contentPic);
+                Glide.with(context).load(content.getUrl()).into(this.contentPic);
+
+                // Allow the User to view a picture in full-screen
+                this.contentPic.setOnClickListener(unusedView -> {
+                    Glide.with(context).load(content.getUrl()).into(imageContainer);
+                    imageContainer.setVisibility(View.VISIBLE);
+
+                    // If we click this new Imageview...
+                    imageContainer.setOnClickListener(unusedView1 -> {
+                        // Get rid of the picture
+                        imageContainer.setImageResource(0);
+                        imageContainer.setVisibility(View.GONE);
+
+                        // No more interaction with the full-screen ImageView
+                        imageContainer.setOnClickListener(null);
+                    });
+                });
+            }
+
+
+            // If this is a Snap content...
+            if (message.isSnap()) {
```

Comment:
This part for handling the snap messages is pretty long. Can you make it a separate method?